src/components/VideoPlayer/VideoPlayer.tsx

```diff
@@ -37,13 +47,32 @@ type progressState = {
 class VideoPlayer extends React.Component<VideoPlayerProps, VideoPlayerState> {
   player: ReactPlayer | null = null;
   state: VideoPlayerState = {
+    ignoreUpdates: true,
     playing: false,
     seeking: false,
     played: 0,
     url: "",
     videoUrlQueue: [],
   };
 
+  constructor(props: VideoPlayerProps) {
+    super(props);
+    const queryString = '?' + props.router.asPath.split('?')[1];
+    const code = new URLSearchParams(queryString).get('code');
```

Comment:
```suggestion
    const { code } = props.router.query;
```
Not sure why, but the above snippet doesn't work. Maybe because I'm refreshing the page?